### ISSUE TITLE
Matplotlib module is not found when using Docker + Jupyter

### ISSUE THREAD

**Issue by vmickus** [devitocodes/devito#1024]
When using Devito with Docker the matplotlib is not being installed.
So out of the box, the tutorials that uses matplolib throws the following error when trying to import it:

`ModuleNotFoundError: No module named 'matplotlib'`

 Creating an empty cell before the usage of matplotlib and installing matplotlib fix the problem:

`!pip install matplotlib`

**Comment by mloubout** [comment]
>Creating an empty cell before the usage

Fixing the docker image would be a better  and cleaner way. This links to #817 and homogenisation of docker between the instruction and pipelines

**Comment by navjotk** [comment]
Yes indeed. Not necessarily related to the linked issue.

**Comment by mloubout** [comment]
Couldn't;t find the issue that goes with the homogenisation sorry linked wrong one.

**Comment by vmickus** [comment]
@navjotk Does adding the requirements-optional to the docker image install is a good option?

What I have tried is to change the following lines from the `dockerfile` file:

```
diff --git a/docker/Dockerfile b/docker/Dockerfile
index 0f0e4e50..c07e9d21 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -5,11 +5,13 @@ RUN apt-get update && apt-get install -y -q \
     libmpich-dev 
 
 ADD ./requirements.txt /app/requirements.txt
+ADD ./requirements-optional.txt /app/requirements-optional.txt
 
 RUN python3 -m venv /venv && \
     /venv/bin/pip install --no-cache-dir --upgrade pip && \
     /venv/bin/pip install --no-cache-dir jupyter && \
-    /venv/bin/pip install --no-cache-dir -r /app/requirements.txt
+    /venv/bin/pip install --no-cache-dir -r /app/requirements.txt && \
+    /venv/bin/pip install --no-cache-dir -r /app/requirements-optional.txt
```

As I have just started using Docker, I am not sure if this is correct. 
And basically this solution involves installing the optional requirements by default. 

IMO if the notebooks need something from the optional requirements to work, it shouldn't be optional.

**Comment by mloubout** [comment]
You can look at the pipeline dockerfile for optional arguments.

**Comment by BlockSprintZIf** [comment]
Dockerfile fixed to add dependencies as mentioned by @vmickus above.

**Comment by mloubout** [comment]
FIxed in #1067
